### ISSUE TITLE
Allow FormControl access to label 'for' Property

### ISSUE THREAD

**Issue by adenta** [uber/baseweb#2285]
It would be nice for accessibility's sake to be able to link labels to inputs within BaseWeb.

**Comment by gergelyke** [comment]
Hi @adenta - it's handled for you automatically. If you go to https://baseweb.design/components/form-control/#examples, and click "Input label", the focus is automatically transferred.

This is what we do under the hood: https://github.com/uber/baseweb/blob/master/src/form-control/form-control.js#L87

Please feel free to reopen if you are looking for something else!

**Comment by jasongi-actu** [comment]
@gergelyke `for` has more uses than just focusing. It reads out the label when using a screen read when you focus on the input element. Without it (or `aria-labelledby`) then you have no idea what the input element is for. I think this issue should be reopened, and there should be a `for` and `id` (so that you can use `aria-labelledby`) prop for the FormControl that are applied to the Label for accessibility purposes.

If you want to try this out for yourself, turn on VoiceOver (if you're using OSX) and run the following snippet in safari ([here is it on w3schools](https://www.w3schools.com/code/tryit.asp?filename=GCC3R2PNY9PI))
```
<label for='label_for'>This label is read out when you focus the input because of the for</label>
<br />
<input type='text' id='label_for'>
<br />
<br />
<label id='label_1'>This label is read out when you focus the input because of the aria-labelledby</label>
<br />
<input type='text' aria-labelledby='label_1'>
<br />
<br />
<label >This label is not read out at all when you focus, all you hear is "edit text"</label>
<br />
<input type='text'>
```

You'll notice that the third input just says "edit text" when you focus into it, while the others read out the label. This is currently what baseweb does - much to the frustration of anyone who uses a screen reader. To be honest I think somehow making these props automatically generated would be a great way of making the web more accessible (given baseweb's popularity), since most developers do not consider accessibility at all and simply exposing the props will mean 95% of developers won't use them.

**Comment by jasongi-actu** [comment]
I've just had a look at the code @gergelyke and it seems like the `for` attribute is set when an id is passed into the child component. This is better than I thought in that initial comment, but still provides no way of using aria-labelledby (which to be able to use you would have to be able to set the id of the label) and doesn't work if there is no id on the child element.

Since a lot of React code I've seen people don't bother setting ids (because accessing DOM elements via ids breaks component abstraction) it would be nice if the FormControl component could set the id if it isn't already set so that it is accessible out of the box.

**Comment by gergelyke** [comment]
@jasongi-actu can you send a proposal implementation in the form of a PR?
